### ISSUE TITLE
Addition assignment operator is broken

### ISSUE THREAD

**Issue by mitsubishirgb** [slint-ui/slint#11843]
### Bug Description

x += y does not trigger animate x, while the equivalent expression x = x + y does.
Both expressions appear semantically equivalent, so I expected animations to behave the same way.

### Reproducible Code (if applicable)

```slint
`import { Button } from "std-widgets.slint";

export component MainWindow inherits Window {
    width: 300px;
    height: 200px;

    r := Rectangle {
        x: 0px;
        width: 50px;
        height: 50px;
        background: green;

        animate x {
            duration: 1s;
        }
    }

    Button {
        clicked => {
            // does NOT animate
            r.x += 50px;

            // DOES animate
            // r.x = r.x + 50px;
        }
    }
}`
```

### Environment Details

- Slint Version: 1.16.1
- Platform/OS: Win 11 
- Programming Language: C++
- Backend/Renderer: Default

**Comment by ogoffart** [comment]
Thanks for your bug report. 

This is a valid bug: the SelfAssignment expression doesn't take into account anomation.
My suggestion to solve this:
 - handle animation in the interpreter with SelfAssignment
 - in the LLR, replace SelfAssignment with an assignment when there is an animation on that property.
